security/advancedtls/advancedtls_test.go

```diff
@@ -679,3 +681,105 @@ func TestOptionsConfig(t *testing.T) {
 		})
 	}
 }
+
+const (
+	tlsEcdheRsaWith3desEdeCbcSha uint16 = 0xc012
+)
+
+const (
+	CurveP256 tls.CurveID = 23
+)
+
+const (
+	pointFormatUncompressed uint8 = 0
+)
+
+const (
+	VersionTLS10 uint16 = 0x0301
+)
+
+func TestGetCertificateSNI(t *testing.T) {
+	serverPeerCert1, err := tls.LoadX509KeyPair(testdata.Path("server_cert_1.pem"),
+		testdata.Path("server_key_1.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	serverPeerCert2, err := tls.LoadX509KeyPair(testdata.Path("server_cert_2.pem"),
+		testdata.Path("server_key_2.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	serverPeerCert3, err := tls.LoadX509KeyPair(testdata.Path("server_cert_3.pem"),
+		testdata.Path("server_key_3.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	tests := []struct {
+		desc          string
+		serverGetCert func(*tls.ClientHelloInfo) ([]*tls.Certificate, error)
+	}{
+		{
+			desc: "Select the certificate that matches the server name provided in clientHello",
+			serverGetCert: func(info *tls.ClientHelloInfo) ([]*tls.Certificate, error) {
+				return []*tls.Certificate{&serverPeerCert1, &serverPeerCert2, &serverPeerCert3}, nil
+			},
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			serverOptions := &ServerOptions{
+				GetCertificate: test.serverGetCert,
+			}
+			serverConfig, err := serverOptions.config()
+			if err != nil {
+				t.Fatalf("Unable to generate serverConfig. Error: %v", err)
+			}
+			// "foo.bar.com" is the common name on server certificate server_cert_1.pem.
```

Comment:
For this, can you please also document in the README file that is used to generate the certificates?

Reply:
Sure! Would you want me to add an appendix section and list the common name for each certificate there?

Reply:
Yes, please! I just thought we probably wanted to document the purposes of all the test certs, since the number could grow larger as time goes. You can do that for the certs you added and I will add them for the rest.

Reply:
Sounds good!